src/app/components/chat/client.ts

```diff
@@ -433,43 +405,39 @@ export function setMessageEditing(chat: ChatClient, message: ChatMessage | null)
 	chat.messageEditing = message;
 }
 
-export function retryFailedQueuedMessage(chat: ChatClient, message: ChatMessage) {
+export function retryFailedQueuedMessage(room: ChatRoom, message: ChatMessage) {
 	if (!message._isQueued || !message._error) {
 		return;
 	}
 
-	// Assign new logged on time and order queued messages by date.
 	message.logged_on = new Date();
+	arrayRemove(room.queuedMessages, i => i.id === message.id);
```

Comment:
Shouldn't we insert this back into the `queuedMessages` array, sorting it to be the newest message?

It looks like this current implementation would remove the message from the messages display, and our message would be lost forever if it failed to send again (or if we couldn't find a room channel for this room).

`sendChatMessage` appears to be removing it from `queuedMessages` after the message is successfully sent anyway, so I think we just need to update `logged_on` and update the index for this message.